src/sentry/migrations/0546_migrate_no_action_dupe_issue_alerts.py

```diff
@@ -0,0 +1,81 @@
+# Generated by Django 3.2.20 on 2023-09-08 20:31
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def has_duplicate_rule(apps, rule_data, project, rule_id):
+    Rule = apps.get_model("sentry", "Rule")
+
+    matchers = {key for key in list(rule_data.keys()) if key not in ("name", "user_id")}
+    extra_fields = ["actions", "environment"]
+    matchers.update(extra_fields)
+    existing_rules = Rule.objects.exclude(id=rule_id).filter(project=project, status=0)
+    for existing_rule in existing_rules:
+        keys = 0
+        matches = 0
+        for matcher in matchers:
+            if existing_rule.data.get(matcher) and rule_data.get(matcher):
+                keys += 1
+
+                if existing_rule.data[matcher] == rule_data[matcher]:
+                    matches += 1
+
+            elif matcher in extra_fields:
+                if not existing_rule.data.get(matcher) and not rule_data.get(matcher):
+                    # neither rule has the matcher
+                    continue
+
+                elif matcher == "environment":
+                    if existing_rule.environment_id and rule_data.get(matcher):
+                        keys += 1
+                        if existing_rule.environment_id == rule_data.get(matcher):
+                            matches += 1
+                    else:
+                        keys += 1
+                else:
+                    # one rule has the matcher and the other one doesn't
+                    keys += 1
+
+        if keys == matches:
```

Comment:
Should this be inside the loop? The function could return on the first iteration of the loop.

Reply:
Isn't it inside the loop? 🤔

Reply:
Or are you saying it shouldn't be? All I care about is if it finds one match.

Reply:
Yep I just need to know if there is a duplicate at all